src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenUsingDeclarationTests.cs

```diff
@@ -1089,5 +1099,168 @@ static async Task Main()
             CompileAndVerify(source, expectedOutput: "Dispose; ");
         }
 
+        [Fact]
+        public void UsingDeclarationAsync()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+class C1 : IAsyncDisposable
+{
+    public ValueTask DisposeAsync() 
```

Comment:
>public ValueTask DisposeAsync()  [](start = 4, length = 32)

Please add a variant of this test with an explicit implementation of IAsyncDisposable.